lib/puppet/type/banner.rb

```diff
@@ -38,6 +48,14 @@
         type:    'String',
         desc:    'The MOTD banner',
       },
+      login:      {
+        type:    'String',
+        desc:    'The Login banner',
+      },
+      exec:      {
+        type:    'String',
```

Comment:
Please make these optional, so that implementations can choose not to implement those.

Reply:
Completely blanked on that sorry.
Updated now.